spec/lib/http/connection_spec.rb

```diff
@@ -62,5 +62,13 @@
       end
       expect(buffer).to eq "1234567890"
     end
+
+    it "fill outbuf when present" do
+      connection.read_headers!
+      outbuf = String.new
+      buffer = String.new
+      buffer << outbuf while connection.readpartial(3, outbuf)
+      expect(buffer).to eq "1234567890"
```

Comment:
Due to the problem above, if you will change size to `1` or `2` you will get corrupted data: `1224447790` or `12345457890` respectively.